kroxylicious-kms-provider-hashicorp-vault/src/test/java/io/kroxylicious/kms/provider/hashicorp/vault/VaultKmsTest.java

```diff
@@ -0,0 +1,91 @@
+import java.net.InetAddress;
+import java.net.InetSocketAddress;
+import java.net.URI;
+import java.net.http.HttpConnectTimeoutException;
+import java.net.http.HttpTimeoutException;
+import java.time.Duration;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+
+import org.junit.jupiter.api.Disabled;
+import org.junit.jupiter.api.Test;
+
+import com.sun.net.httpserver.HttpServer;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+class VaultKmsTest {
+
+    public static final URI NON_ROUTABLE = URI.create("http://10.255.255.1/");
+
+    @Test
+    public void testConnectionTimeout() {
+        VaultKms kms = new VaultKms(NON_ROUTABLE, "token", Duration.ofMillis(500));
+        CompletableFuture<String> alias = kms.resolveAlias("alias");
+        assertThat(alias).failsWithin(Duration.ofSeconds(2))
+                .withThrowableOfType(ExecutionException.class)
+                .withCauseInstanceOf(HttpConnectTimeoutException.class);
+    }
+
+    @Test
+    public void testRequestTimeoutWaitingForHeaders() {
+        HttpServer httpServer = delayResponse(Duration.ofSeconds(1), Duration.ZERO);
+        URI address = URI.create("http://" + httpServer.getAddress().getHostName() + ":" + httpServer.getAddress().getPort() + "/");
+        VaultKms kms = new VaultKms(address, "token", Duration.ofMillis(500));
+        CompletableFuture<String> alias = kms.resolveAlias("alias");
+        assertThat(alias).failsWithin(Duration.ofSeconds(2))
+                .withThrowableOfType(ExecutionException.class)
+                .withCauseInstanceOf(HttpTimeoutException.class);
+        httpServer.stop(0);
+    }
+
+    @Disabled // JDK http client request timeout cancelled after headers received currently, todo add our own timeout future
+    @Test
+    public void testRequestTimeoutWaitingForBody() {
+        HttpServer httpServer = delayResponse(Duration.ZERO, Duration.ofSeconds(1));
+        URI address = URI.create("http://" + httpServer.getAddress().getHostName() + ":" + httpServer.getAddress().getPort() + "/");
+        VaultKms kms = new VaultKms(address, "token", Duration.ofMillis(500));
+        CompletableFuture<String> alias = kms.resolveAlias("alias");
+        assertThat(alias).failsWithin(Duration.ofSeconds(2))
+                .withThrowableOfType(ExecutionException.class)
+                .withCauseInstanceOf(HttpTimeoutException.class);
+        httpServer.stop(0);
+    }
+
+    public static HttpServer delayResponse(Duration delayHeaders, Duration delayResponseAfterHeaders) {
+        try {
+            HttpServer server = HttpServer.create(new InetSocketAddress(InetAddress.getLocalHost(), 0), 0);
+            server.createContext("/", exchange -> {
+                sleep(delayHeaders);
+                exchange.sendResponseHeaders(200, 1);
+                sleep(delayResponseAfterHeaders);
+                exchange.getResponseBody().write(5);
+                exchange.close();
```

Comment:
```suggestion
                var buf = new byte[] { 5 };
                exchange.sendResponseHeaders(200, buf.length);
                sleep(delayResponseAfterHeaders);
                exchange.getResponseBody().write(buf);
                exchange.close();
```

Reply:
have extracted a constant with naming to show it's arbitrary